### DISCUSSION TITLE
TypeScript: Option 'noEmit' cannot be specified with option 'composite'

### DISCUSSION THREAD

**Discussion by s-panferov** [vercel/next.js#26429] [Ideas]
# Bug report

## Describe the bug

TypeScript error: 

```
Type error: Option 'noEmit' cannot be specified with option 'composite'.
```

Project's tsconfig.json:

```json
{
	"compilerOptions": {
               // only this is relevant
               "composite": true
         },
	"include": ["src/**/*"],
	"exclude": ["node_modules"]
}
```

## To Reproduce

Create a TypeScript project with a next.js project as a composite

## Expected behavior

NextJS always wants to insert `noEmit` option to my tsconfig, but it conflicts with `composite` option I'm using. Setting `noEmit` to `false` does not help. Potential solutions:

1. Detect the `composite` option
2. Allow to bypass the `tsconfing.json` modification (which I think was already proposed but declined)

## System information

- TypeScript: 3.9.7
- next: 9.5.2

**Comment by cheapCoder** [comment]
any update? And why this issue was moved to discussion
